src/Files.App/Data/Commands/Manager/CommandManager.cs

```diff
@@ -420,9 +423,51 @@ private void OverwriteKeyBindings()
 				}
 			}
 
-			_allKeyBindings = commands.Values
-				.SelectMany(command => command.HotKeys, (command, hotKey) => (Command: command, HotKey: hotKey))
-				.ToImmutableDictionary(item => item.HotKey, item => item.Command);
+			try
+			{
+				_allKeyBindings = commands.Values
+					.SelectMany(command => command.HotKeys, (command, hotKey) => (Command: command, HotKey: hotKey))
+					.ToImmutableDictionary(item => item.HotKey, item => item.Command);
+			}
+			catch (ArgumentException ex)
+			{
+				// The keys are not necessarily all different because they can be set manually in text editor
+				// ISSUE: https://github.com/files-community/Files/issues/15331
+
+				var flat = commands.Values.SelectMany(x => x.HotKeys).Select(x => x.LocalizedLabel);
+				var duplicates = flat.GroupBy(x => x).Where(x => x.Count() > 1).Select(group => group.Key);
+
+				foreach (var item in duplicates)
+				{
+					if (!string.IsNullOrEmpty(item))
+					{
+						var occurrences = allCommands.Where(x => x.HotKeys.Select(x => x.LocalizedLabel).Contains(item));
+
+						// Restore the defaults for all occurrences
+						for (int index = 0; index < occurrences.Count(); index++)
+						{
+							occurrences.ElementAt(index).RestoreKeyBindings();
+						}
+					}
+				}
+
+				_allKeyBindings = commands.Values
+					.SelectMany(command => command.HotKeys, (command, hotKey) => (Command: command, HotKey: hotKey))
+					.ToImmutableDictionary(item => item.HotKey, item => item.Command);
+
+				App.Logger.LogError(ex, "The app is temporarily using default key for some commands bindings because of a serious error of assigning custom keys.");
```

Comment:
* Couldn't you write that the reason is that there are duplicate key assigns?
* I think the log level for exceptions with known causes should be Warning, not Error.

Reply:
Done